integrations/nielsen-dtvr/lib/index.js

```diff
@@ -0,0 +1,347 @@
+
+NielsenDTVR.prototype.videoAdStarted = function(track) {
+  var metadata = this.mapAd(track);
+  this.client.ggPM('loadMetadata', metadata);
+  this.sendID3(track);
+};
+
+/**
+ * Video Ad Completed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoAdCompleted = function(track) {
+  this.end(track);
+};
+
+/**
+ * Video Playback Interrupted
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackInterrupted = function(track) {
+  this.sendID3(track);
+  this.end(track);
+};
+
+/**
+ * Video Playback Seek Started
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackSeekStarted = function(track) {
+  this.sendID3(track);
+  this.end(track);
+};
+
+/**
+ * Video Playback Seek Completed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackSeekCompleted = function(track) {
+  var metadata = this.isAd ? this.mapAd(track) : this.mapVideo(track);
+  this.client.ggPM('loadMetadata', metadata);
+  this.sendID3(track);
+};
+
+/**
+ * Video Playback Buffer Started
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackBufferStarted = function(track) {
+  this.sendID3(track);
+  this.end(track);
+};
+
+/**
+ * Video Playback Buffer Completed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackBufferCompleted = function(track) {
+  var metadata = this.isAd ? this.mapAd(track) : this.mapVideo(track);
+  this.client.ggPM('loadMetadata', metadata);
+  this.sendID3(track);
+};
+
+/**
+ * Video Playback Paused
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackPaused = function(track) {
+  this.sendID3(track);
+  this.end(track);
+};
+
+/**
+ * Video Playback Resumed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackResumed = function(track) {
+  var metadata = this.isAd ? this.mapAd(track) : this.mapVideo(track);
+  this.client.ggPM('loadMetadata', metadata);
+  this.sendID3(track);
+};
+
+/**
+ * Video Playback Completed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackCompleted = function(track) {
+  this.end(track);
+};
+
+/**
+ * Send ID3 tags to Nielsen
+ *
+ * @api private
+ */
+
+NielsenDTVR.prototype.sendID3 = function(event) {
+  var id3Events = this.options.sendId3Events;
+  var i;
+  var id3Prop;
+  var id3Tags;
+
+  if (id3Events.length <= 0) return;
+  if (id3Events.length > 0) {
+    for (i = 0; i < id3Events.length; i += 1) {
+      if (id3Events[i] === event.event()) {
+        break;
+      }
+      if (i === id3Events.length - 1) {
+        return;
+      }
+    }
+  }
+
+  id3Prop = this.options.id3Property || 'ID3';
+  id3Tags = event.proxy('properties.' + id3Prop);
+  if (id3Tags) {
+    // we'll only send ID3 tags to Nielsen if we detect the customer has either
+    // never sent ID3 tags, or if they're different from the previous ID3 tags
+    // sent during the session
+    if (!this.ID3) {
+      this.ID3 = id3Tags;
+      this.client.ggPM('sendID3', this.ID3);
+    } else if (id3Tags !== this.ID3) {
+      this.client.ggPM('sendID3', this.ID3);
```

Comment:
This would be easier to understand if we do something like:
```javascript
if (!this.ID3 || id3Tags !== this.ID3) {
  this.ID3 = id3Tags;
}
this.client.ggPM('sendID3', this.ID3);
```